dvc/progress.py

```diff
@@ -131,3 +134,40 @@ def format_dict(self):
             d["ncols_desc"] = 1
             d["prefix"] = ""
         return d
+
+
+# Adding this here for showcase only
+from contextlib import contextmanager  # noqa
+
+
+class ContextFlags:
+    def __init__(self):
+        self._stack = []
```

Comment:
why `stack` isn't a `set`? if the idea is to do:

```
return name if name in stack else None
```

I think finding values on a set is constant